src/Vibrato/Vibrato.cpp

```diff
@@ -0,0 +1,48 @@
+#include <cassert>
+
+#include "Vibrato.h"
+#include "LFO.h"
+
+Vibrato::Vibrato(float sampleRate, float maxDepth, int numChannels = 1)
+: sampleRate(sampleRate), lfo(sampleRate), maxDepth(maxDepth) {
+    assert(numChannels > 0);
+    for (int i = 0; i < numChannels; i++) {
+        int length = (int)ceilf(sampleRate * maxDepth * 2) + 2;
+        delayLines.emplace_back(new RingBuffer<float>(length));
+    }
+}
+
+void Vibrato::setFrequency(float frequency) {
+    lfo.setFrequency(frequency);
+}
+
+Error_t Vibrato::setDepth(float depth) {
+    if (depth < -maxDepth || depth > maxDepth) {
+        return Error_t::kFunctionInvalidArgsError;
+    }
+    lfo.setAmplitude(depth * sampleRate);
+    return Error_t::kNoError;
+}
+
+float Vibrato::getFrequency() const {
+    return lfo.getFrequency();
+}
+
+float Vibrato::getDepth() const {
+    return lfo.getAmplitude() / sampleRate;
+}
+
+void Vibrato::process(float **input, float **output, int numFrames) {
+    const float baseDelay = maxDepth * sampleRate + 1;
+    for (int i = 0; i < numFrames; i++) {
+        const float mod = lfo.process();
```

Comment:
I like this naming scheme where you have to do vibrato.process and lfo.process to call the process functions! Makes it readable and not seeming like a inheritance problem like I have seen in other projects.